xocto/ranges.py

```diff
@@ -965,6 +965,32 @@ def days(self) -> int:
         return (self.end - self.start).days + 1
 
 
+def finite_date_range_from_str(period_str: str) -> FiniteDateRange:
```

Comment:
❓ How about making this a class method of `Range`?

```python
Range.parse(...)
```

This is a little more complicated, but makes sense to me as a complement to the existing `__str__` method, which renders the same format:

https://github.com/octoenergy/xocto/blob/49a1a8313a4e9b50b4f504f70ca192156f66f82c/xocto/ranges.py#L227-L238